File: ws/gudino/u2/ws12Inheritance/Farmsystem/src/main/java/ec/edu/espe/farmsymulator/model/Cow.java
package ec.edu.espe.farmsymulator.model;

import java.util.Date;

/**
 *
 * @author Bryan Gudino
 */
public class Cow extends FarmAnimal {
    private boolean isProducingMilk;
    private float litersADay;

    public Cow(boolean isProducingMilk, float litersADay, int id, String breed, Date bornOn, String gender, boolean isAbleToReproduce, float weight, Cage cage) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        this.isProducingMilk = isProducingMilk;
        this.litersADay = litersADay;
    }

    public float milk() {
        if (isProducingMilk) {
            return litersADay;
        } else {
            return 0.0f;
        }
    }

    @Override
    public String toString() {
        return "Cow{" + "isProducingMilk=" + isProducingMilk + ", litersADay=" + litersADay + "," + super.toString() + '}';
    }

    // Getters and Setters
    /**
     * @return the isProducingMilk
     */
    public boolean isIsProducingMilk() {
        return isProducingMilk;
    }

    /**
     * @param isProducingMilk the isProducingMilk to set
     */
    public void setIsProducingMilk(boolean isProducingMilk) {
        this.isProducingMilk = isProducingMilk;
    }

    /**
     * @return the litersADay
     */
    public float getLitersADay() {
        return litersADay;
    }

    /**
     * @param litersADay the litersADay to set
     */
    public void setLitersADay(float litersADay) {
        this.litersADay = litersADay;
    }
}

File: hw/loza/u2/hw19Interface/Python/model/Cow.py
from model.FarmAnimal import FarmAnimal

class Cow(FarmAnimal):
    def __init__(self, isProducingMilk, litersADay, id, breed, bornOn, gender, isAbleToReproduce, weight, cage):
        super().__init__(id, breed, bornOn, gender, isAbleToReproduce, weight, cage)
        self.isProducingMilk = isProducingMilk
        self.litersADay = litersADay

    def __str__(self):
        return f"""{{
        "isProducingMilk": {self.isProducingMilk},
        "litersADay": {self.litersADay},
        "farmAnimal": {super().__str__()}
        }}"""

    def is_is_producing_milk(self):
        return self.isProducingMilk

    def set_is_producing_milk(self, isProducingMilk):
        self.isProducingMilk = isProducingMilk

    def get_liters_a_day(self):
        return self.litersADay

    def set_liters_a_day(self, litersADay):
        self.litersADay = litersADay
